rq/worker.py

```diff
@@ -946,17 +948,20 @@ def prepare_job_execution(self, job: 'Job'):
         """Performs misc bookkeeping like updating states prior to
         job execution.
         """
-
+        self.log.debug(f"Preparing for execution of Job ID {job.id}")
         with self.connection.pipeline() as pipeline:
             self.set_current_job_id(job.id, pipeline=pipeline)
             self.set_current_job_working_time(0, pipeline=pipeline)
+            self.log.debug(f"Set Current Job ID & Working Time")
 
             heartbeat_ttl = self.get_heartbeat_ttl(job)
             self.heartbeat(heartbeat_ttl, pipeline=pipeline)
             job.heartbeat(utcnow(), heartbeat_ttl, pipeline=pipeline)
+            self.log.debug(f"Got Heartbeat. Current TTL is {heartbeat_ttl}")
 
             job.prepare_for_execution(self.name, pipeline=pipeline)
             pipeline.execute()
+            self.log.debug(f"Job preparation finished.")
```

Comment:
Actually I just realized that the commands for these three log statements are executed in a single `pipeline.execute()` call, meaning when the log is output, the current job ID, working time and heartbeat have not been set yet.

This means we should only log after a `pipeline.execute()`

Reply:
Not sure I understand how you'd like this to be. My initial idea was just to set logs to understand what was the last active point before a potential issue - those particular logs don't necessarily depend on the pipeline execution. Anyways, fixed the logging text. This is the only one missing, let me know how you'd like this to be.

Reply:
Sorry, then let's just delete these two lines:
1. `self.log.debug(f"Set Current Job ID & Working Time")`
2. `self.log.debug(f"Heartbeat set with TTL of {heartbeat_ttl}")`

I'll merge this PR in after these two are removed.